### ISSUE TITLE
Support ES2015 modules

### ISSUE THREAD

**Issue by p-bakker** [mozilla/rhino#945]
Big topic, see [MDN Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) for some primer

**Comment by gausie** [comment]
I'd like to be able to continue to specify a resolver as we do for the existing require implementation

**Comment by p-bakker** [comment]
I agree: where we use Rhino, I foresee us loading modules from a blob in the database or some other persistence store, so we need full control over how module paths are resolved

**Comment by p-bakker** [comment]
@szegedi as you authored the CommonJS implementation at one point in time and [you seem to like fun tasks](https://github.com/mozilla/rhino/issues/1444#issuecomment-2191020287) (😛), any chance you'd be interested in taking a stab at this one? 🙏

**Comment by Rhython** [comment]
Is there any progress on this issue? Currently I run es modules by preprocessing them into commonjs modules. It would be nice if this could be supported natively.
